### PR TITLE
chore: reduce create pipeline validation cognitive complexity and format frontend code

### PR DESCRIPTION
**Pull request by chenhaiyun**
*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

## Summary

(describe what this merge request does)

## Implementation highlights

(describe how the merge request does for feature changes, share the RFC link if it has)

## Test checklist

- [ ] add new test cases
- [ ] all code changes are covered by unit tests
- [ ] end-to-end tests
  - [ ] deploy web console with CloudFront + S3 + API gateway
  - [ ] deploy web console within VPC
  - [ ] deploy ingestion server
    - [ ] with MSK sink
    - [ ] with KDS sink
    - [ ] with S3 sink
  - [ ] deploy data processing
  - [ ] deploy data modeling
    - [ ] new Redshift Serverless
    - [ ] provisioned Redshift
    - [ ] Athena
  - [ ] deploy with reporting

## Is it a breaking change

- [ ] add parameters without default value in stack
- [ ] introduce new service permission in stack
- [ ] introduce new top level stack module

## Miscellaneous

- [ ] introduce new symbol link source file(s) to be shared among infra code, web console frontend, and web console backend

### CODE REVIEW

**Comment by chenhaiyun** [comment]
@tyyzqmf  We can run `npm run format` before commit codes if the **auto save and format** are not working.

**Comment by zxkane** [comment]
> @tyyzqmf We can run `npm run format` before commit codes if the **auto save and format** are not working.

You can also add the command to `git commit` hook.

**Review by zxkane — CHANGES_REQUESTED** [review]
Also update ci script to validate the code format,

https://github.com/awslabs/clickstream-analytics-on-aws/blob/73a1e174a4e4cbcc0b29c7773d0751540a539fc9/.github/workflows/build-sub-projects.yml#L17-L21

https://github.com/awslabs/clickstream-analytics-on-aws/blob/73a1e174a4e4cbcc0b29c7773d0751540a539fc9/.github/workflows/build.yml#L74-L78

**Comment by tyyzqmf** [comment]
> @tyyzqmf We can run `npm run format` before commit codes if the **auto save and format** are not working.

If run format in frontend, please reset *-ln.ts files.

**Comment by chenhaiyun** [comment]
> > @tyyzqmf We can run `npm run format` before commit codes if the **auto save and format** are not working.
> 
> If run format in frontend, please reset `*-ln.ts` files OR ignore `*-ln.ts` files .

added `.prettierignore` to ignore these files

File: .github/workflows/build-sub-projects.yml [hunk 1]
```diff
@@ -17,6 +17,7 @@ jobs:
       - run: |-
           cd frontend/
           yarn install --check-files --frozen-lockfile
+          yarn run lint
```
Comment (zxkane) [inline .github/workflows/build-sub-projects.yml:20]: please add a step to fail the build if finding the mutations,

https://github.com/awslabs/clickstream-analytics-on-aws/blob/73a1e174a4e4cbcc0b29c7773d0751540a539fc9/.github/workflows/build.yml#L74-L78

**Review by zxkane — CHANGES_REQUESTED** [review]

File: .github/workflows/build-sub-projects.yml (hunk 1 above)
Comment (zxkane) [reply .github/workflows/build-sub-projects.yml:20]:
```
    - git diff --staged --patch --exit-code > .repo.patch || export mutations_happened=true
    - if [ "$mutations_happened" = "true" ]; then cat .repo.patch && exit 1; fi;
 ```

**Review by zxkane — APPROVED** [review]

**Review by tyyzqmf — APPROVED** [review]

**Pull request merged**